js 3/main.js

```diff
@@ -0,0 +1,239 @@
+    console.log(`Mostly I wash my car once a ${this.term}, but now I do It every ${this.currentTerm}.`);
+  }
+};
+
+car.introduce();
+car.displayFeatures();
+car.washCar();
+
+let pet = {
+  name: 'Max',
+  type: 'Dog',
+  breed: 'Golden Retriever',
+  age: 5,
+  color: 'golden',
+  favoriteActivities: ['fetch', 'swimming', 'long walks'],
+  introduce() {
+    console.log(`This is ${this.name}, my ${this.color} ${this.breed} ${this.type}.`);
+  },
+  displayActivities() {
+    console.log(`${this.name}'s favorite activities are:`);
+    this.favoriteActivities.forEach((activity, index) => {
+      console.log(`${index + 1}. ${activity}`);
+    });
+  },
+  celebrateBirthday() {
+    this.age++;
+    console.log(`${this.name} just turned ${this.age} years old. Happy birthday!`);
+  }
+};
+
+pet.introduce();
+pet.displayActivities();
+pet.celebrateBirthday();
+
+let laptop = {
+  brand: 'Apple',
+  model: 'MacBook Pro',
+  year: 2023,
+  color: 'silver',
+  age: 2,
+  specifications: {
+    processor: 'Intel Core i7',
+    memory: '16GB',
+    storage: '512GB SSD'
+  },
+  introduce() {
+    console.log(`I own a ${this.color} ${this.brand} ${this.model} (${this.year}).`);
+  },
+  displaySpecifications() {
+    console.log('Specifications:');
+    for (let key in this.specifications) {
+      console.log(`${key}: ${this.specifications[key]}`);
+    }
+  },
+  displayAge() {
+    console.log(`I have used this ${this.brand} ${this.model} for ${this.age} years.`);
+  }
+};
+
+laptop.introduce();
+laptop.displaySpecifications();
+laptop.displayAge();
+*/
+//////////////////////////////////////////////////////////
+
+// 6. 
+/*
+let securedSelfIntroduce = car.introduce.bind(car);
+let securedSelfDisplayFeatures = car.displayFeatures.bind(car);
+let securedSelfWashCar = car.washCar.bind(car);
+
+setTimeout(securedSelfIntroduce, 1000);
+setTimeout(securedSelfDisplayFeatures, 2000);
+setTimeout(securedSelfWashCar, 3000);
+
+let securedSelfIntroduce1 = pet.introduce.bind(pet);
+let securedSelfDisplayActivities = pet.displayActivities.bind(pet)
+let securedSelfCelebrateBirthday = pet.celebrateBirthday.bind(pet)
+
+setTimeout(securedSelfIntroduce1, 1000);
+setTimeout(securedSelfDisplayActivities, 2000);
+setTimeout(securedSelfCelebrateBirthday, 3000);
+
+let securedSelfIntroduce2 = laptop.introduce.bind(laptop);
+let securedSelfDisplaySpecifications = laptop.displaySpecifications.bind(laptop);
+let securedSelfDisplayAge = laptop.displayAge.bind(laptop);
+
+setTimeout(securedSelfIntroduce2, 1000);
+setTimeout(securedSelfDisplaySpecifications, 2000);
+setTimeout(securedSelfDisplayAge, 3000);
+*/
+//////////////////////////////////////////////////////////
+
+// 7. 
+
+function sum(a, b) {
+  let result = a + b;
+  console.log(`Sum: ${result}`);
+}
+
+function slower(func, seconds) {
+  return function (...args) {
+    console.log(`Chill out, you will get your result in ${seconds} seconds.`);
+    setTimeout(function () {
+      func.apply(null, args);
```

Comment:
```suggestion
    setTimeout(() => func.apply(this, args);
```

щоб не було втрати контексту нам якраз треба передавати this